packages/core-data/src/reducer.js

```diff
@@ -214,6 +289,66 @@ export const entities = ( state = {}, action ) => {
 	};
 };
 
+/**
+ * Reducer keeping track of entity edit undo history.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+const UNDO_INITIAL_STATE = [];
+UNDO_INITIAL_STATE.offset = 0;
+export function undo( state = UNDO_INITIAL_STATE, action ) {
+	switch ( action.type ) {
+		case 'EDIT_ENTITY_RECORD':
```

Comment:
Something to think about for later maybe: This could be made more generic. Instead of checking the action type, we could rely on the presence of `undo` property in the `meta` key  in order to support any random action.

Reply:
Yeah, that would be nice. We can add that if it's needed.